### PR TITLE
Support both ghc-8.10 and ghc-9.2

### PR DESCRIPTION
**Pull request by erikd**
This should not be merged until https://github.com/well-typed/canonical-json/pull/8 is merged and a new version of `canonical-json` is released. When that is done the commit "Temporarily use a patched version of canonical-json" can be dropped.

### CODE REVIEW

File: cardano-prelude-test/test/Test/Cardano/Prelude/GHC/Heap/SizeSpec.hs [hunk 1]
```diff
@@ -103,6 +105,13 @@ mkBS :: [Word8] -> ByteString
 {-# NOINLINE mkBS #-}
 mkBS = BS.Strict.pack
 
+adjustBsSize :: NumWords -> NumWords
```
Comment (angerman) [inline cardano-prelude-test/test/Test/Cardano/Prelude/GHC/Heap/SizeSpec.hs:108]: can we get a comment on _why_ we have `-1` now?

Comment (erikd) [reply cardano-prelude-test/test/Test/Cardano/Prelude/GHC/Heap/SizeSpec.hs:108]: Done!

File: cabal.project [hunk 2]
```diff
@@ -3,3 +3,8 @@ index-state: 2022-07-01T00:00:00Z
 packages:
   cardano-prelude
   cardano-prelude-test
+
+source-repository-package
+  type: git
+  location: https://github.com/well-typed/canonical-json
+  tag: 8a37c22f0ce4759b4d534a4019a473e8225f3477
```
Comment (LudvikGalois) [inline cabal.project:10]: You need to throw a --sha comment here for Nix

Comment (erikd) [reply cabal.project]: That has been dropped in favor of a Hackage dependency.

File: cardano-prelude-test/cardano-prelude-test.cabal [hunk 3]
```diff
@@ -79,15 +86,20 @@ test-suite cardano-prelude-test-suite
...
                       -fno-warn-missing-safe-haskell-mode
                       -fno-warn-prepositive-qualified-module
+                      -fno-warn-all-missed-specialisations
+                      -fno-warn-missing-deriving-strategies
+                      -fno-warn-missing-import-lists
+                      -fno-warn-missing-safe-haskell-mode
+                      -fno-warn-prepositive-qualified-module
+                      -fno-warn-safe
+                      -fno-warn-unsafe
+
```
Comment (angerman) [inline cardano-prelude-test/cardano-prelude-test.cabal:102]: Do we still need all of these?

**Review by angerman — APPROVED** [review]
LGTM. I'm not sure we need all the `no-warn` things?
Should we bump the version in a separate PR to
```
version:              0.1.1.0
```

Do we know who's responsible for cutting releases?

File: cardano-prelude-test/cardano-prelude-test.cabal (hunk 3 above)
Comment (erikd) [reply cardano-prelude-test/cardano-prelude-test.cabal:102]: No we do not. I have removed them.

**Pull request merged**